avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -280,16 +281,41 @@ private void processRecord(JVar recordSchemaVar, String recordName, final Schema
         );
       } else {
         JInvocation finalNewRecordInvocation = newRecord;
+        JClass finalRecordClass = recordClass;
         ifCodeGen(methodBody,
             reuseVar.ne(JExpr._null()),
-            thenBlock -> thenBlock.assign(result, JExpr.cast(recordClass, reuseVar)),
+            thenBlock -> thenBlock.assign(result, JExpr.cast(finalRecordClass, reuseVar)),
             elseBlock -> elseBlock.assign(result, finalNewRecordInvocation)
         );
       }
     } else {
       result = null;
     }
+
+    int fieldCount = 0;
+    int fieldsWrittenToTarget = 0;
+    JBlock popMethodBody = methodBody;
+    JMethod popMethod = null;
     for (Schema.Field field : recordWriterSchema.getFields()) {
+      // We roll the population method for very large records, the initial fields are kept in the outer method as original to maintain performance for smaller records
+      fieldCount++;
+      if (fieldCount % FIELDS_PER_POPULATION_METHOD == 0) {
+        popMethod = generatedClass.method(JMod.PUBLIC, codeModel.VOID,
+                getUniqueName("populate_" + recordName));
+
+        popMethod._throws(IOException.class);
+        if (recordAction.getShouldRead()) {
+          popMethod.param(recordClass, recordName);
```

Comment:
It is not clear to me whether the new smaller methods are using object reuse or not.
As @FelixGV mentioned, please commit the generated classes as well.

Reply:
@gaojieliu can you give me example of such test that checks the actual generated code?
In terms of reuse, yes it should work fine since the reuse is use at the start to decide which object to populate and the smaller functions just do the actual population

Reply:
This is an example:

https://github.com/linkedin/avro-util/commit/c52235837db3954fd1c2a5671e0ccb4e5ed9f3ac

The commits before that are the changes related to this code gen.

Reply:
But this is just the code gen, I haven't seen a test that actually validates the generated code itself, do you want me to just commit the generated code with a lower threshold? in that case I just wonder how I should configure the threshold? constructor?

Reply:
If you look at the commits preceding the one I linked, you will see the tests which validate that fast-avro works.

The goal of committing the code-gen is to be able to easily visualize the diff in code-gen output associated with a change in the code-gen logic. And we split off the code-gen output into a standalone commit to make it easier to sort through the logic changes versus code-gen output changes (the PR UI lets you pick a specific commit).

Reply:
As for configurability, I leave that up to your judgment. A telescopic constructor, or system property or a package private setter which the test can access by being in the same package... any of these could do the trick.

Reply:
Hmm, I see a lot of tests for correctness of the deserializer but not something checking the code itself, maybe you can give me an example of such Test function? I have looked in the previous commits in the PR as well...
I will add the configuration anyway and I will make the tests work with a low value (e.g. 2) so all the functionality will be tested with this thing in place, however the default value for normal usage should probably be something like 100. 
In that case the generated code will indeed be different and I can commit that as well, would that be sufficient?

Reply:
Yes, I think that's sufficient.